ee/tasks/subscriptions/slack_subscriptions.py

```diff
@@ -0,0 +1,113 @@
+from typing import Dict, List
+
+import structlog
+from django.conf import settings
+
+from posthog.models.exported_asset import ExportedAsset
+from posthog.models.integration import Integration, SlackIntegration
+from posthog.models.subscription import Subscription
+
+logger = structlog.get_logger(__name__)
+
+UTM_TAGS_BASE = "utm_source=posthog&utm_campaign=subscription_report"
+
+
+def _block_for_asset(asset: ExportedAsset) -> Dict:
+    image_url = asset.get_public_content_url()
+    alt_text = None
+    if asset.insight:
+        alt_text = asset.insight.name or asset.insight.derived_name
+
+    if settings.DEBUG:
+        image_url = "https://source.unsplash.com/random"
+
+    return {"type": "image", "image_url": image_url, "alt_text": alt_text}
+
+
+def send_slack_subscription_report(
+    subscription: Subscription, assets: List[ExportedAsset], total_asset_count: int,
+) -> None:
+    utm_tags = f"{UTM_TAGS_BASE}&utm_medium=slack"
+
+    resource_name = None
```

Comment:
These three things always travel together. Could they come as a dataclass from a method on Subscription?

Reply:
That's a great idea 👍